azure-pipelines.yml

```diff
@@ -42,32 +42,103 @@ variables:
+        steps:
+        - task: PowerShell@2
+          displayName: 'Install Windows SDK'
+          inputs:
+            targetType: filePath
+            filePath: './eng/install-windows-sdk.ps1'
+            failOnStderr: true
+            showWarnings: true
+
+        - task: PowerShell@2
+          displayName: 'Install procdump'
+          inputs:
+            targetType: filePath
+            filePath: ./eng/install-procdump.ps1
+            failOnStderr: true
+            showWarnings: true
+
+        - task: PowerShell@2
+          displayName: 'Install Access Database Engine'
+          inputs:
+            targetType: filePath
+            filePath: ./eng/install-access-database-engine.ps1
+            failOnStderr: true
+            showWarnings: true
+
+        # Build only production projects, no tests
+        - script: eng\common\CIBuild.cmd
+            -configuration Release
+            -prepareMachine
+            /p:Publish=false
+            /p:Test=false
+          name: BuildAndPack
+          displayName: Build and Pack
+
+        - task: PublishBuildArtifacts@1
+          displayName: 'Publish NuGet packages for testing'
+          inputs:
+            PathtoPublish: '$(Build.SourcesDirectory)/artifacts/packages/Release'
+            ArtifactName: 'NuGetPackages'
+          condition: always()
 
+- stage: TestWindows
+  displayName: Test Windows
+  dependsOn: BuildAndPack
+  jobs:
   - template: /eng/common/templates/jobs/jobs.yml
     parameters:
       enableMicrobuild: true
       enablePublishBuildArtifacts: true
       enablePublishTestResults: true
       testResultsFormat: 'vstest'
-      enablePublishBuildAssets: true
-      enablePublishUsingPipelines: true
+      enablePublishBuildAssets: false
+      enablePublishUsingPipelines: false
       enableTelemetry: true
       jobs:
-      - job: Windows
+      - job: Windows_Debug
+        displayName: Windows Debug Tests
         timeoutInMinutes: 90
         pool:
           name: NetCore-Public
           demands: ImageOverride -equals windows.vs2022preview.amd64.open
-        strategy:
-          matrix:
-            Release:
-              _BuildConfig: Release
-            Debug:
-              _BuildConfig: Debug
         steps:
+        - task: DownloadBuildArtifacts@0
+          displayName: 'Download NuGet packages'
+          inputs:
+            buildType: 'current'
+            downloadType: 'single'
+            artifactName: 'NuGetPackages'
+            downloadPath: '$(System.DefaultWorkingDirectory)'
+          condition: ne('${{ parameters.SkipTests }}', 'True')
+
+        - task: CopyFiles@2
+          displayName: 'Copy packages to expected location'
+          inputs:
+            SourceFolder: '$(System.DefaultWorkingDirectory)/NuGetPackages'
+            Contents: '**'
+            TargetFolder: '$(Build.SourcesDirectory)/artifacts/packages/Release'
+          condition: ne('${{ parameters.SkipTests }}', 'True')
```

Comment:
Can you move all the parameters.SkipTests conditions to the job level instead of being on every task?